### PR TITLE
feat: add custom claims support

### PR DESCRIPTION
**Pull request by nikhil-stytch**
- Add custom claims to authenticate requests & responses
- Custom claims will not be visible on the session object until the product is officially released

### CODE REVIEW

**Review by jeremy-stytch — COMMENTED** [review]
LGTM after some fixes and a question:

If someone upgrades to this new version and uses a `session_custom_claims` request field, how does the API handle it? I would still approve the PR, but I don't want to merge+publish this yet if it the API isn't ready for it (errors or other weirdness).

File: lib/shared.ts [hunk 1]
```diff
@@ -291,6 +291,7 @@ export interface Session {
   expires_at: Date;
   attributes: Attributes;
   authentication_factors: AuthenticationFactor[];
+  custom_claims?: Map<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
```
[the remark this answers is not in the document]
Comment (nikhil-stytch) [reply lib/shared.ts]: Thanks! Updated here [e4b77ba](https://github.com/stytchauth/stytch-node/pull/146/commits/e4b77ba82a4ccdab149f81cb069283240ebe3170)

File: lib/crypto_wallets.ts [hunk 2]
```diff
@@ -29,6 +29,7 @@ export interface AuthenticateRequest {
   session_token?: string;
   session_jwt?: string;
   session_duration_minutes?: number;
+  session_custom_claims?: Map<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
```
[the remark this answers is not in the document]
Comment (nikhil-stytch) [reply lib/crypto_wallets.ts]: Thanks! Updated here [e4b77ba](https://github.com/stytchauth/stytch-node/pull/146/commits/e4b77ba82a4ccdab149f81cb069283240ebe3170)

**Comment by nikhil-stytch** [comment]
> LGTM after some fixes and a question:
> 
> If someone upgrades to this new version and uses a `session_custom_claims` request field, how does the API handle it? I would still approve the PR, but I don't want to merge+publish this yet if it the API isn't ready for it (errors or other weirdness).

The API handles this situation gracefully but does not attach custom claims to sessions in responses until we fully release this product. This behavior is consistent across client libraries and cURL'ing

**Review by jeremy-stytch — APPROVED** [review]

**Pull request merged**